app/src/main/java/com/github/multimatum_team/multimatum/viewmodel/GroupViewModel.kt

```diff
@@ -0,0 +1,97 @@
+package com.github.multimatum_team.multimatum.viewmodel
+
+import android.app.Application
+import android.util.Log
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.github.multimatum_team.multimatum.model.GroupID
+import com.github.multimatum_team.multimatum.model.UserGroup
+import com.github.multimatum_team.multimatum.repository.AuthRepository
+import com.github.multimatum_team.multimatum.repository.DeadlineID
+import com.github.multimatum_team.multimatum.repository.GroupRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+/**
+ * ViewModel for the group view.
+ * Acts as a bridge between the group repository (Firebase) and the user interface.
+ * The ViewModel ensures that the data is always fresh, keeping the database and the UI in sync.
+ * The list of groups is also updated when the GroupRepository notifies that the authenticated
+ * user has changed.
+ */
+@HiltViewModel
+class GroupViewModel @Inject constructor(
```

Comment:
This class is completly untested. I don't think it is a good idea to make the tests this sprint with the work you have already done but it can be a good issue for a next sprint

Reply:
I will test it in this PR, it's not too complicated (mostly the same as the other viewmodel tests)